index.ts

```diff
@@ -53,10 +53,14 @@ export const recommendationsToHTML = (
 <div class="${classes.wrapper}">
   <h1 class="${classes.header}">Recommended Articles</h1>
 ${recommendations.reduce(
-  (output, { title, category, uri, imageUri }) =>
+  (output, { title, category, uri, imageUri }, index) =>
     `${output}
     
-  <a href="${uri}" class="${classes.card}">
+  <a href="${uri}" class="${classes.card}" id="${classPrefix}-recommendation-${
+      index + 1
+    }" onclick="window.dataLayer?.push({ event:'recommendation-click', recommendationNumber: ${
```

Comment:
I assume the onclick is fired before the actual navigation occurs?  Were you able to test locally to see if it works?

Reply:
I did no testing. I don't have have any analytics on my test page since it's meant to be placed in AEM. I could try but I figured we could just test it in AEM.

Reply:
Gotcha, yeah we can test and see if this works with both a href and an onclick. Another solution was to just have the onclick and create a function that makes the push to the dataLayer and then navigates to the page.